write-fonts/src/write.rs

```diff
@@ -43,7 +43,10 @@ pub fn dump_table<T: FontWrite + Validate>(table: &T) -> Result<Vec<u8>, Validat
     let mut writer = TableWriter::default();
     table.write_into(&mut writer);
     let mut graph = writer.finish();
-    graph.topological_sort();
+    if !graph.topological_sort() {
+        //TODO: implement extension promotion and table splitting
+        panic!("could not find a graph packing that handles all offsets, aborting");
```

Comment:
feels a touch weird to panic in a function that returns a Result?

Reply:
The error returned by this function contains information about any invalid data in the table that would prevent us from being able to compile it (for instance a vec with more than u16::max items) but a failure to pack the table is something else entirely.

It's possible that this should be rethought? Even if we implement all the fanciness, we probably can't *guarantee* that table packing will always succeed, so maybe this should be fallible..